packages/gasket-core/lib/index.d.ts

```diff
@@ -1,21 +1,115 @@
-import { Gasket, GasketConfigDefinition } from '@gasket/engine';
-
-declare module '@gasket/engine' {
+declare module '@gasket/core' {
+  export type MaybeMultiple<T> = T | Array<T>;
+  export type MaybeAsync<T> = T | Promise<T>;
+  export type ResolvedType<T> = T extends Promise<infer Value> ? Value : T;
 
   export interface GasketActions {}
 
-  export interface GasketConfig {}
+  // To be extended by plugins
+  export interface HookExecTypes {}
 
-  export interface Gasket {
-    new (config: GasketConfigDefinition): Gasket
-    actions: GasketActions
-  }
 
-  export interface HookExecTypes {
+  export interface CoreHookExecTypes {
```

Comment:
I believe this needs stay `HookExecTypes` in order for `init`, etc to be available has `HookHandler`s
```suggestion
  // To be extended by plugins
  export interface HookExecTypes {
```

You can get rid of the empty declaration above it

Reply:
@agerard-godaddy when I remove the empty declaration 10+ packages fail their typecheck. Would it be ok to leave this as is and create a new ticket to fix the types?

Reply:
I pushed some fixes per slack